app/models/spree/variant_decorator.rb

```diff
@@ -84,6 +84,54 @@ def self.indexed
     ]
   end
 
+  # Overriding `Spree::Variant.on_hand` in old Spree versions.
+  # It doesn't exist in newer Spree versions.
+  def on_hand
+    if respond_to? :total_on_hand
+      # This is Spree 2.0
+      total_on_hand
+    elsif Spree::Config[:track_inventory_levels] && !on_demand
+      count_on_hand
+    else
+      Float::INFINITY
+    end
+  end
+
+  # Overriding `Spree::Variant.on_hand=` in old Spree versions.
+  # It doesn't exist in newer Spree versions.
+  def on_hand=(new_level)
+    error = 'Cannot set on_hand value when Spree::Config[:track_inventory_levels] is false'
+    raise error unless Spree::Config[:track_inventory_levels]
+
+    self.count_on_hand = new_level unless on_demand
```

Comment:
:warning: this is directly affected by https://github.com/openfoodfoundation/spree/pull/8

Reply:
Ah, that's where I saw this code before! Thank you. So this pull request would override Frank's change.